packages/@aws-cdk/aws-kms/lib/key.ts

```diff
@@ -384,25 +411,49 @@ export class Key extends KeyBase {
     this.keyId = resource.ref;
     resource.applyRemovalPolicy(props.removalPolicy);
 
+    (props.admins ?? []).forEach((p) => this.grantAdmin(p));
+
     if (props.alias !== undefined) {
       this.addAlias(props.alias);
     }
   }
 
-  private allowAccountIdentitiesToControl() {
+  /**
+   * Grant admins permissions using this key to the given principal
+   *
+   * Key administrators have permissions to manage the key (e.g., change permissions, revoke), but do not have permissions
+   * to use the key in cryptographic operations (e.g., encrypt, decrypt).
+   */
+  public grantAdmin(grantee: iam.IGrantable): iam.Grant {
+    return this.grant(grantee, ...perms.ADMIN_ACTIONS);
+  }
```

Comment:
Should this be moved up, where the other `grant*()` methods are defined? (And also addd to `IKey`?)

Reply:
This was intentional, cautious decision, although I can be convinced otherwise.

With imported keys, the key policy itself can't be altered, meaning that only the IAM principal policy will be edited. This effectively means for the `grant` to work, the current CDK role needs to be a key admin already. This is workable for places where a user has imported a key from the same environment, but any cross-account usage won't work; to set up cross-account admin, it really has to be done on the key policy itself (for a `Key`, not `IKey`). I also think the number of use cases for adding admins from another context/construct are far, far, fewer than adding key users.

The README already contains several disclaimers about how calling `grant` on an imported Key may (or may not) do what you expect, and pulling this up into `IKey` would be a backwards-compatible enhancement, so I thought I'd "play it safe" and start out with it here before moving it up (based on customer feedback).